src/rgw/rgw_kms.cc

```diff
@@ -688,24 +688,24 @@ class TransitSecretEngine: public VaultSecretEngine {
     int res = send_request(dpp, "POST", "", config_path,
                            post_data, y, dummy_bl);
     if (res < 0) {
-      return res;
-    }
-    if (dummy_bl.length() != 0) {
       ldpp_dout(dpp, 0) << "ERROR: unexpected response from Vault marking key to delete: "
-	<< dummy_bl
-	<< dendl;
-      return -EINVAL;
+        << dummy_bl.c_str()
```

Comment:
dummy_bl c_str() strings are not guaranteed to have a terminating nul.  You must also call dummy_bl.length() and probably construct a string view.